File: problem/Delete_Columns_Sorted/minDeletionSize.c
bool isUnsorted(char **strs, int strsSize, int j)
{
    for (int i = 1; i < strsSize; i++)
    {
        if (strs[i][j] < strs[i - 1][j])
            return true;
    }
    return false;
}

int minDeletionSize(char **strs, int strsSize)
{
    int res = 0;
    int len = strlen(strs[0]);
    for (int j = 0; j < len; j++)
    {
        if (isUnsorted(strs, strsSize, j))
            res++;
    }
    return res;
}

File: problem/Delete_Columns_Sorted/minDeletionSize.py
class Solution:
    def minDeletionSize(self, strs):
        def isUnsorted(j):
            for i in range(1, len(strs)):
                if strs[i][j] < strs[i - 1][j]:
                    return True
            return False
        return sum(isUnsorted(j) for j in range(len(strs[0])))